pkg/shell/indexes.js

```diff
@@ -550,7 +550,17 @@ function MachinesIndex(index_options, machines, loader, mdialogs) {
             $("#machine-spinner").toggle(frame && !$(frame).attr("data-ready"));
             $(frame).css('display', 'block');
             $(frame).attr('data-active', 'true');
-            item = compiled.items[state.component];
+
+            let component_manifest = state.component;
+            // If `state.component` is not known to any manifest, find where it comes from
+            if (compiled.items[state.component] == undefined) {
```

Comment:
`== undefined` is rather sloppy, as e.g. `null == undefined` is true. If you really *only* want `undefined`, use `===`. If you just care about falsy (as I think is more appropriate here), use `!compiled.items[state.component]`.

Reply:
I copied this from 8155456.
It really just needs to check, that accessing of attribute was not successful. I want empty string to not satisfy this condition.
```
undefined == undefined
true
undefined == null
true
undefined == ""
false
```

But it is true, that accessing non-existing element returns undefined:
```
x={a:1}
Object { a: 1 }

x.foo
undefined
```

I'll update to use `===` in both places.
```